### ISSUE TITLE
False positive E0001 for CodeUri in AWS::Serverless::Function

### ISSUE THREAD

**Issue by corey-cole** [aws-cloudformation/cfn-lint#2658]
### CloudFormation Lint Version

cfn-lint 0.76.1

### What operating system are you using?

Mac

### Describe the bug

cfn-lint gives a false-positive error when CodeUri is an S3 URI.

```E0001 Error transforming template: Resource with id [rCustomAuthorizerFunction] is invalid. 'CodeUri' requires Bucket and Key properties to be specified.```

Not only does this template deploy via CloudFormation, the warning from cfn-lint contradicts the [public documentation for this resource type](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-function.html#Examples)

### Expected behavior

cfn-lint should not give a false positive that contradicts public documentation

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Transform:
  - 'AWS::LanguageExtensions'
  - 'AWS::Serverless-2016-10-31'

Description: cfn-lint test case

Mappings:
  RegionMap:
    us-east-1:
      InsightsArn: arn:aws:lambda:us-east-1:580247275435:layer:LambdaInsightsExtension:21
    us-west-2:
      InsightsArn: arn:aws:lambda:us-west-2:580247275435:layer:LambdaInsightsExtension:21

Resources:
  rCustomAuthorizerFunction:
    Type: AWS::Serverless::Function
    Properties:
      FunctionName: !Sub "my-custom-authorizer-${AWS::Region}"
      Description: Custom authorizer for APIGW
      Runtime: nodejs16.x
      Handler: index.handler
      CodeUri: !Sub "s3://custom-authz-${AWS::AccountId}-${AWS::Region}/function-v1.0.zip"
      Policies:
        - arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy
        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
      Layers:
        - !FindInMap [RegionMap, !Ref AWS::Region, "InsightsArn"]

```
